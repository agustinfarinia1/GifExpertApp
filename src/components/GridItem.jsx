
import PropTypes from 'prop-types';

const GridItem = ({id,title,url}) => {
    return (
        <div className="card">
            <img src={url} alt={title}></img>
            <p>{title}</p>
        </div>
    );
}

GridItem.propTypes={
    title : PropTypes.string.isRequired,
    url : PropTypes.string.isRequired,
}

export default GridItem;