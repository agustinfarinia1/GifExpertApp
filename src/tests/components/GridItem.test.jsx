import { render, screen } from "@testing-library/react";
import GridItem from "../../components/GridItem";

describe('Prueba Componente <GridItem/>',() => {
    
    const title= "hola";
    const url = "https://hola/prueba.com"

    test('prueba 01- el campo title y url debe ser isRequired', () => {
        const {container} = render(<GridItem title={title} url={url}/>);
        expect(container).toMatchSnapshot();
    })

    test('prueba 02- Tiene que coincidir title y url', () => {
        render(<GridItem title={title} url={url}/>);
        const {src,alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    })

    test('prueba 03- Debe mostrar el titulo en el componente', () => {
        render(<GridItem title={title} url={url}/>);
        expect(screen.getByText(title).innerHTML).toContain(title);
    })
})