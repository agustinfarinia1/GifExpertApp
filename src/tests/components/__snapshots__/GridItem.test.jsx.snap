// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba Componente <GridItem/> prueba 01- el campo title y url debe ser isRequired 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="hola"
      src="https://hola/prueba.com"
    />
    <p>
      hola
    </p>
  </div>
</div>
`;
